### PR TITLE
Fix stream methods

### LINKED ISSUE #367: Doc Update required for get_activity_streams()
The documentation for get_activity_streams() at [https://stravalib.readthedocs.io/en/latest/reference/api/stravalib.client.Client.get_activity_streams.html](https://stravalib.readthedocs.io/en/latest/reference/api/stravalib.client.Client.get_activity_streams.html) seems not accurate any longer. The resolution 'all' seems not to exist any longer, using it explicitly results in a Bad Request. In contrast, 'high' seems to have a higher resolution than 10.000 points as I received a response with 11.500 points having a resolution 'high'.

### LINKED ISSUE #383: Method to get stream kinda broken.
The method to get the streams (`get_activity_streams`, `get_effort_streams`, `get_segment_streams`) seems to not match the schema from strava https://developers.strava.com/playground/#/Streams/getActivityStreams

The types are supposed to be passed as a query argument and not as part of the path. Also, this argument which they call `key` in the schema is required.

I suggest we remove the default of None for those functions and fix the way we build the request. I can take care of it if you agree with the proposed changes.

### PR DESCRIPTION
**Pull request by enadeau**
closes #383 and closes #367

## Description

Update the client's methods to get stream to match the current strata API.

## Type of change

Select the statement best describes this pull request.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This is a documentation update
- [ ] Other (please describe)

## Does your PR include tests

If you are fixing a bug or adding a feature, we appreciate (but do not require)
tests to support whatever fix of feature you're implementing.

- [x] Yes
- [ ] No, i'd like some help with tests
- [ ] This change doesn't require tests

## Did you include your contribution to the change log?

- [x] Yes, `changelog.md` is up-to-date.

*If you are having a hard time getting the tests to run correctly feel free to
ping one of the maintainers here!*

### CODE REVIEW

File: stravalib/client.py [hunk 1]
```diff
@@ -1383,54 +1397,46 @@ def get_activity_streams(
...
-        # Pack streams into dictionary
-        try:
-            return {i.type: i for i in streams}
-        except exc.ObjectNotFound:
-            return None  # just to be explicit.
+        print(response)
+        return {
+            model.Stream.parse_obj(stream)
+            for stream_type, stream in response.json().items()
+        }
```
Comment (jsamoocha) [inline stravalib/client.py:1439]: This is much cleaner than the existing approach but is a breaking change.

File: stravalib/client.py [hunk 2]
```diff
@@ -31,6 +31,20 @@
...
+    "altitude",
+    "velocity_smooth",
+    "heartrate",
+    "cadence",
+    "watts",
+    "temp",
+    "moving",
+    "grade_smooth",
+]
+
```
Comment (jsamoocha) [inline stravalib/client.py:47]: We can use `strava_model.StreamType.schema()['enum']` to get these values from the OpenAPI spec.

File: stravalib/client.py (hunk 1 above)
Comment (enadeau) [reply stravalib/client.py:1439]: My bad I intended to do
```py
return {
    stream_type: model.Stream.parse_obj(stream)
    for stream_type, stream in response.json().items()
}
```
but I should probably also get the None back if an exception is raised

File: stravalib/client.py (hunk 2 above)
Comment (enadeau) [reply stravalib/client.py:47]: Oh that's really nice.

**Comment by enadeau** [comment]
I've written some test to bring the coverage to 100% on the method.

I think it is ready for review

The `get_effrot_streams` and `get_segments_streams` still need to be modified, but the change would be identical to the one I did for `get_activity_streams` so best to get the review in first.

**Review by jsamoocha — COMMENTED** [review]
Thanks for this fix and the added test coverage! Because Strava still supports these undocumented parameters, we cannot simply remove them, as that would change the behavior of our API.

Besides that, your overall approach of using the documented query params for the stream types instead of the path param and getting rid of the unnecessary `BatchedResultsIterator` is good and can be applied for other endpoints involving streams.

**Comment by enadeau** [comment]
Applied your suggestions and also did the effort and segment method. 

I've refactored the code to avoid having the same logic 3 time.

Ready for review

**Review by jsamoocha — APPROVED** [review]
Thanks a lot, the refactoring and added test coverage are great improvements!

**Pull request merged**